### PR TITLE
Add RBAC to satellite daemonset

### PR DESCRIPTION
**Pull request by jbanety**
Lokomotive clusters have PodSecurityPolicy (PSP) enabled by default so I need to apply to satellite containers a permissive PSP.

### CODE REVIEW

**Comment by kvaps** [comment]
Thanks, wouldn't it work with the [`automountServiceAccountToken: false`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server)?

**Comment by jbanety** [comment]
Nop. 
We need a service account to assign permissions to access host network, run root containers,...
And to enforce security, Lokomotive has a webhook to disable automounting default service account.
https://kinvolk.io/docs/lokomotive/0.5/concepts/admission-webhook/

**Comment by kvaps** [comment]
Got it! Could you also contribute PSP policy itself?

Example manifest from old stable repo
https://github.com/helm/charts/blob/2f030a8e9db3e451e8030b12d2658371f5708f7d/stable/elasticsearch/templates/podsecuritypolicy.yaml

**Comment by jbanety** [comment]
Haha, I have already created _static_ manifests in my Lokomotive component.
I still have to define the permissions to apply.
Containers are running as root. 
Can we run them as regular user ?

**Comment by kvaps** [comment]
Well it's not a problem to run controller and other daemons as unprivileged user. But I'm not sure if it is possible to make satellite daemon running, because it is operating with the LVM, devicemapper, DRBD and ZFS quite hard.
I'm worry that specific permissions are required for it.

**Comment by kvaps** [comment]
Thanks! I'll review and merge this in a while!🤘

**Comment by jbanety** [comment]
This is a very permissive policy.
I did not find time to dig into this.

**Comment by kvaps** [comment]
I think you don't have to. PSP is going to be deprecated in the next Kubernetes release:
https://github.com/kubernetes/kubernetes/pull/97171

**Comment by jbanety** [comment]
OK. That's enough then.

**Comment by jbanety** [comment]
Sorry for the mess. This is OK now :roll_eyes:

File: helm/kube-linstor/templates/controller-rbac.yaml [hunk 1]
```diff
@@ -16,6 +16,10 @@ rules:
   - apiGroups: [""]
     resources: ["endpoints", "endpoints/restricted"]
     verbs: ["create", "patch", "update"]
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["{{ $fullName }}"]
+    verbs: ["use"]
```
Comment (kvaps) [inline helm/kube-linstor/templates/controller-rbac.yaml:23]:
```suggestion
  {{- if .Values.podSecurityPolicy.enabled }}
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["{{ $fullName }}"]
    verbs: ["use"]
  {{- end }}
```

File: helm/kube-linstor/templates/csi-controller-rbac.yaml [hunk 2]
```diff
@@ -74,6 +74,10 @@ rules:
     resources: ["leases"]
     verbs: ["get", "update"]
     resourceNames: ["linstor-csi-linbit-com"]
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["{{ $fullName }}"]
+    verbs: ["use"]
```
Comment (kvaps) [inline helm/kube-linstor/templates/csi-controller-rbac.yaml:81]:
```suggestion
  {{- if .Values.podSecurityPolicy.enabled }}
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["{{ $fullName }}"]
    verbs: ["use"]
    {{- end }}
```

File: helm/kube-linstor/templates/podsecuritypolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      min: 0
+  privileged: true
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+    - '*'
```
Comment (kvaps) [inline helm/kube-linstor/templates/podsecuritypolicy.yaml]:
```suggestion
{{- $fullName := include "linstor.fullname" . -}}
{{- if .Values.podSecurityPolicy.enabled }}
apiVersion: policy/v1beta1
kind: PodSecurityPolicy
metadata:
  name: {{ $fullName }}
  annotations:
    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
spec:
  allowPrivilegeEscalation: true
  allowedCapabilities:
    - '*'
  fsGroup:
    rule: RunAsAny
  hostIPC: true
  hostNetwork: true
  hostPID: true
  hostPorts:
    - max: 65535
      min: 0
  privileged: true
  runAsUser:
    rule: RunAsAny
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    rule: RunAsAny
  volumes:
    - '*'
{{- end }}
```

File: helm/kube-linstor/templates/stork-scheduler-rbac.yaml [hunk 4]
```diff
@@ -58,6 +58,10 @@ rules:
   - apiGroups: ["coordination.k8s.io"]
     resources: ["leases"]
     verbs: ["get", "create", "update"]
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["{{ $fullName }}"]
+    verbs: ["use"]
```
Comment (kvaps) [inline helm/kube-linstor/templates/stork-scheduler-rbac.yaml:65]:
```suggestion
  {{- if .Values.podSecurityPolicy.enabled }}
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["{{ $fullName }}"]
    verbs: ["use"]
  {{- end }}
```

File: helm/kube-linstor/templates/satellite-daemonset.yaml [hunk 5]
```diff
@@ -131,6 +131,7 @@ spec:
       hostPID: true
       imagePullSecrets:
         {{- toYaml .Values.satellite.image.pullSecrets | nindent 8 }}
+      serviceAccountName: {{ $fullName }}-satellite-sa
```
Comment (kvaps) [inline helm/kube-linstor/templates/satellite-daemonset.yaml:135]:
```suggestion
      {{- if .Values.podSecurityPolicy.enabled }}
      serviceAccountName: {{ $fullName }}-satellite-sa
      {{- end }}
```

File: helm/kube-linstor/templates/satellite-rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{{- $fullName := include "linstor.fullname" . -}}
+{{- if .Values.satellite.enabled }}
```
Comment (kvaps) [inline helm/kube-linstor/templates/satellite-rbac.yaml]:
```suggestion
{{- if and .Values.satellite.enabled .Values.podSecurityPolicy.enabled }}
```

**Review by kvaps — COMMENTED** [review]
Let's make this configurable, also add to values.yaml

```yaml
## Specify if a Pod Security Policy for node-exporter must be created
## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/
##
podSecurityPolicy:
  enabled: false
```

File: helm/kube-linstor/templates/csi-node-rbac.yaml [hunk 7]
```diff
@@ -9,6 +9,10 @@ rules:
   - apiGroups: [""]
     resources: ["events"]
     verbs: ["get", "list", "watch", "create", "update", "patch"]
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["{{ $fullName }}"]
+    verbs: ["use"]
```
Comment (kvaps) [inline helm/kube-linstor/templates/csi-node-rbac.yaml:16]:
```suggestion
  {{- if .Values.podSecurityPolicy.enabled }}
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["{{ $fullName }}"]
    verbs: ["use"]
  {{- end }}
```

**Review by kvaps — COMMENTED** [review]
forgot these two

File: helm/kube-linstor/templates/ha-controller-rbac.yaml [hunk 8]
```diff
@@ -25,6 +25,14 @@ rules:
...
     verbs: ["get", "update"]
     resourceNames: ["{{ $fullName }}-ha-controller"]
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["{{ $fullName }}"]
+    verbs: ["use"]
+  - apiGroups: ["extensions"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["{{ $fullName }}"]
+    verbs: ["use"]
```
Comment (kvaps) [inline helm/kube-linstor/templates/ha-controller-rbac.yaml:36]:
```suggestion
  {{- if .Values.podSecurityPolicy.enabled }}
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["{{ $fullName }}"]
    verbs: ["use"]
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["{{ $fullName }}"]
    verbs: ["use"]
  {{- end }}
```

**Comment by kvaps** [comment]
merged, thanks!

**Pull request merged**